### PR TITLE
Enums for Sense Data

### PR DESCRIPTION
**Pull request by basimsahaf**
This PR contains enums for Sense Response and Sense Key as mentioned in #5 

What to review in this PR:
- SenseResponse Enum
- SenseKey Enum

### CODE REVIEW

File: lib/scsi_defs.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    CONDITION_MET = "04h",
+    BUSY = "08h",
+    INTERMEDIATE = "10h",               // obsolete
+    INTERMEDIATE_CONDITION_MET = "14h", // obsolete
+    RESERVATION_CONFLICT = "18h",
+    COMMAND_TERMINATED = "22h",         // obsolete
+    TASK_SET_FULL = "28h",
+    ACA_ACTIVE = "30h",
+    TASK_ABORTED = "40h",
+  };
```
Comment (basimsahaf) [inline lib/scsi_defs.h:18]: Old code from https://github.com/googleinterns/scsi2nvme/pull/16

**Comment by eyhhuang** [comment]
Do you mind rebasing off `basic-enums` again so the changes are a bit easier to see?

**Comment by basimsahaf** [comment]
@eyhhuang you should be able to see the changes more clearly now

**Comment by basimsahaf** [comment]
Marking this PR as draft for now as I am waiting on SenseKey Code Qualifiers

File: lib/scsi_defs.h [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+    ControlByte(uint8_t obsolete, uint8_t naca, uint8_t reserved, uint8_t vendorSpecific) : 
+      obsolete{obsolete}, naca{naca}, reserved{reserved}, vendorSpecific{vendorSpecific} {}
+
+    uint8_t getValue() const {
+      return 0 | obsolete | (naca << 2) | (reserved << 3) | (vendorSpecific << 6);
+    }
+
+    // Getters
+    uint8_t getObsolete() const {
```
Comment (w-michal) [inline lib/scsi_defs.h:119]: Please follow what was done in second PR - we don't need getters/seters for "basic" objects.

File: lib/scsi_defs.h [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+  class ControlByte {
+   private:
+    uint8_t obsolete : 2;
+    uint8_t naca : 1;
+    uint8_t reserved : 3;
+    uint8_t vendorSpecific : 2;
+
+   public:
+    ControlByte() :
```
Comment (w-michal) [inline lib/scsi_defs.h:108]: ctors are not required - zero initialization can be used for first one and second is redundant if we can use aggregate initialization (https://en.cppreference.com/w/cpp/language/aggregate_initialization). We can also make all the fields public.

File: lib/scsi_defs.h [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    CURRENT_FIXED_ERROR = 0x70,
+    DEFERRED_FIXED_ERROR = 0x71
+  };
+
+
+  // Refer to https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf , Section 2.4.1.5 Table 28
+  enum SenseKey {
+    NO_SENSE = 0x0,
+    RECOVERED_ERROR = 0x1,
+    NOT_READY = 0x2,
```
Comment (w-michal) [inline lib/scsi_defs.h:20]: Please follow the same style as other PR. (enum class and kNoSense instead of NO_SENSE).

File: lib/scsi_defs.h (hunk 2 above)
Comment (basimsahaf) [reply lib/scsi_defs.h]: That was actually the old code from #16 

I re-based and this PR now shows the updated code

File: lib/scsi_defs.h (hunk 3 above)
Comment (basimsahaf) [reply lib/scsi_defs.h]: Same as https://github.com/googleinterns/scsi2nvme/pull/17#discussion_r426740888

File: lib/scsi_defs.h (hunk 4 above)
Comment (basimsahaf) [reply lib/scsi_defs.h]: Updated to `enum class` and changed the style for enum values

File: lib/scsi_defs.h [hunk 5]
```diff
@@ -80,5 +80,33 @@ namespace scsi_defs {
     uint8_t vendorSpecific : 2;
   } ControlByte;
 
+  // Refer to https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf , Section 2.4.1 Table 11
+  enum class SenseResponse: uint8_t {
+    kCurrentDescriptorError = 0x72,
+    kDeferredDescriptorError = 0x73,
+    kCurrentFixedError = 0x70,
+    kDeferredFixedError = 0x71
```
Comment (eyhhuang) [inline lib/scsi_defs.h:88]: nit: order the enums by value

File: lib/scsi_defs.h [hunk 6]
```diff
@@ -80,5 +80,33 @@ namespace scsi_defs {
     uint8_t vendorSpecific : 2;
   } ControlByte;
 
+  // Refer to https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf , Section 2.4.1 Table 11
+  enum class SenseResponse: uint8_t {
```
Comment (eyhhuang) [inline lib/scsi_defs.h:84]: nit: Code in master currently has a space before the `:`

**Pull request merged**